Replace custom method with File.ReadAllText() in ScriptAnalysis.cs

amritanand-py opened the pull request:
Fix #25136

The ReadScript helper method in src/System.Management.Automation/engine/Modules/ScriptAnalysis.cs was redundant. It created a FileStream and StreamReader only to return the file’s contents, which can be achieved directly with File.ReadAllText(path, Encoding.Default).

This PR removes the ReadScript method and replaces all call sites with File.ReadAllText.

Changes

Removed the ReadScript method.

Updated all usages to:

`File.ReadAllText(path, Encoding.Default)`

<!-- Anything that looks like this is a comment and can't be seen after the Pull Request is created. -->

# PR Summary
This PR simplifies file reading in ScriptAnalysis.cs by removing the unnecessary ReadScript helper method. The built-in File.ReadAllText method provides the same functionality in a cleaner and more efficient way.

Removed:
ReadScript method and replaces all call sites with File.ReadAllText.
<!-- Summarize your PR between here and the checklist. -->

## PR Context
Extra code in ScriptAnalysis.cs #25136


<!-- Provide a little reasoning as to why this Pull Request helps and why you have opened it. -->

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge. If this PR is a work in progress, please open this as a [Draft Pull Request and mark it as Ready to Review when it is ready to merge](https://docs.github.com/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests).
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
  - [x] None
  - **OR**
  - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.5/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
    - [ ] Experimental feature name(s): <!-- Experimental feature name(s) here -->
- **User-facing changes**
  - [x] Not Applicable
  - **OR**
  - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [ ] Issue filed: <!-- Number/link of that issue here -->
- **Testing - New and feature**
  - [x] N/A or can only be tested interactively
  - **OR**
  - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

amritanand-py committed:
Remove: unused SafeFileHandle in  src\System.Management.Automation\engine\Modules\ScriptAnalysis.cs

The ReadScript method was retrieving a SafeFileHandle from the FileStream but never using it.
This line was unnecessary since proper disposal is already handled by the using blocks.

Changes

Removed redundant SafeFileHandle assignment.

xtqqczze commented:
This may be a change in behavior, e.g. if the `SafeFileHandle` getter throws.

amritanand-py commented:
Yes, removing the SafeFileHandle line does technically change behavior, because the getter could throw in some scenarios (e.g., if the stream was already disposed or in restricted environments).

However, since the handle value was never used, this call didn’t contribute to the method’s functionality.

The method now only depends on FileStream and StreamReader, both of which are already safely wrapped in using.

xtqqczze commented:
The same functionality could be implemented with [File.ReadAllText Method](https://learn.microsoft.com/dotnet/api/system.io.file.readalltext).

amritanand-py commented:
I will update this PR with Refactor commit

amritanand-py commented:
well there is two approach for this 

internal static string ReadScript(string path)
{
    return File.ReadAllText(path, Encoding.UTF8);
}


or directly change and
eradicate method itself since it holds only 2 references and optimizing one step further

Before 
            string dataFileContents = ScriptAnalysis.ReadScript(psDataFilePath);

After
             string dataFileContents = File.ReadAllText(psDataFilePath, Encoding.Default);

amritanand-py committed:
Merge branch 'PowerShell:master' into master

amritanand-py committed:
Remove ReadScript method and use File.ReadAllText directly

The ReadScript helper method has been removed, and all call sites have been updated to use File.ReadAllText(path, Encoding.Default) directly.
Changes

Deleted the ReadScript method from ScriptAnalysis.cs.

Replaced all usages of ReadScript with:
File.ReadAllText(path, Encoding.Default)

amritanand-py commented:
@microsoft-github-policy-service agree

xtqqczze commented:
@amritanand-py Please update the PR description.

amritanand-py commented:
@xtqqczze Done

iSazonov commented on src/System.Management.Automation/engine/Modules/ScriptAnalysis.cs:
```diff
@@ -89,19 +89,6 @@ internal static ScriptAnalysis Analyze(string path, ExecutionContext context)
             return result;
         }
 
-        internal static string ReadScript(string path)
```
@MartinGC94 Do you think it is good to remove the method? Won't this code be less readable?

MartinGC94 replied:
I think it's perfectly fine. If anything it's arguably easier to understand when it's just `File.ReadAllText` because you don't have to check if `ReadScript` does anything weird.

iSazonov commented:
/azp run PowerShell-Windows-Packaging-CI, PowerShell-CI-linux-packaging

iSazonov committed:
Merge branch 'master' into master

iSazonov commented:
/azp run PowerShell-Windows-Packaging-CI, PowerShell-CI-linux-packaging

iSazonov commented:
@amritanand-py Please add dummy commit to unlock merging.

xtqqczze commented:
> @amritanand-py Please add dummy commit to unlock merging.

You can add an empty commit using:

```sh
git commit --allow-empty --message 'Empty commit'
```

amritanand-py committed:
Empty commit

iSazonov commented:
/azp run PowerShell-Windows-Packaging-CI, PowerShell-CI-linux-packaging
